RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTEleL1TrackIsolProducer.cc

```diff
@@ -0,0 +1,95 @@
+// Author: Sam Harper (RAL/CERN)
+// L1 track isolation producer for the HLT
+// A quick primer on how the E/gamma HLT works w.r.t to ID variables
+// 1. the supercluster is the primary object
+// 2. superclusters get id variables associated to them via association maps keyed
+//    to the supercluster
+// However here we also need to read in electron objects as we need to solve for which
+// GsfTrack associated to the supercluster to use for the isolation
+// The electron producer solves for this and assigns the electron the best GsfTrack
+// which we will use for the vz of the electron
+// One thing which Swagata Mukherjee pointed out is that we have to be careful of
+// is getting a bad GsfTrack with a bad vertex which  will give us a fake vz which then
+// leads to a perfectly isolated electron as that random vz is not a vertex
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/EgammaCandidates/interface/Electron.h"
+#include "DataFormats/EgammaCandidates/interface/ElectronFwd.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateFwd.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateIsolation.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+
+#include "RecoEgamma/EgammaIsolationAlgos/interface/EgammaL1TkIsolation.h"
+
+#include <memory>
+
+class EgammaHLTEleL1TrackIsolProducer : public edm::global::EDProducer<> {
+public:
+  explicit EgammaHLTEleL1TrackIsolProducer(const edm::ParameterSet&);
+  ~EgammaHLTEleL1TrackIsolProducer() override;
```

Comment:
just a suggestion:
```suggestion
  ~EgammaHLTEleL1TrackIsolProducer() override = default;
```
and drop the implementation below.

Reply:
thanks, done, of course this is better :)